relay-statsd/src/lib.rs

```diff
@@ -225,13 +250,12 @@ where
     F: FnOnce(&MetricsClient) -> R,
     R: Default,
 {
-    CURRENT_CLIENT.with(|client| {
-        if let Some(client) = client {
-            f(client)
-        } else {
-            R::default()
-        }
-    })
+    let guard = METRICS_CLIENT.read();
+    if let Some(client) = guard.deref() {
+        f(client)
+    } else {
+        R::default()
+    }
```

Comment:
I am not sure about the performance impact of holding the `guard` here. The previous implementation with thread local `CURRENT_CLIENT` did not allow setting the `MetricsClient` more than once, as the test `current_client_is_global_client` shows.